File: rb/features.rb
# Dexpaprika SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DexpaprikaFeatures
  def self.make_feature(name)
    case name
    when "base"
      DexpaprikaBaseFeature.new
    when "test"
      DexpaprikaTestFeature.new
    else
      DexpaprikaBaseFeature.new
    end
  end
end

File: py/features.py
# Dexpaprika SDK feature factory

from feature.base_feature import DexpaprikaBaseFeature
from feature.test_feature import DexpaprikaTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DexpaprikaBaseFeature(),
        "test": lambda: DexpaprikaTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
